drivers/display/display_st7789v.c

```diff
@@ -63,33 +63,36 @@ struct st7789v_data {
 static void st7789v_set_lcd_margins(const struct device *dev,
 				    uint16_t x_offset, uint16_t y_offset)
 {
-	struct st7789v_data *data = (struct st7789v_data *)dev->data;
+	struct st7789v_data *data = dev->data;
 
 	data->x_offset = x_offset;
 	data->y_offset = y_offset;
 }
 
 static void st7789v_set_cmd(const struct device *dev, int is_cmd)
 {
-	struct st7789v_config *config = (struct st7789v_config *)dev->config;
+	const struct st7789v_config *config = dev->config;
 
 	gpio_pin_set_dt(&config->cmd_data_gpio, is_cmd);
 }
 
 static void st7789v_transmit(const struct device *dev, uint8_t cmd,
-			     uint8_t *tx_data, size_t tx_count)
+			     const uint8_t *tx_data, size_t tx_count)
 {
-	struct st7789v_config *config = (struct st7789v_config *)dev->config;
+	const struct st7789v_config *config = dev->config;
 
-	struct spi_buf tx_buf = { .buf = &cmd, .len = 1 };
-	struct spi_buf_set tx_bufs = { .buffers = &tx_buf, .count = 1 };
+	{
```

Comment:
Why these changes, that bring no benefits at all except more code,  and why are they part of this commit?

Reply:
The order of the commit was a mess, because that was easier then rewrite the history. I fixed that now.

`tx_data` needs to be const, because some of that data now comes from `st7789v_config`. 
`tx_buf` needs to be const, because it holds `tx_data`.
The double code is because you can't write to the const `tx_buf`.
I added a new scope, to prevent shadowing the the `tx_buf` name.
I hope this makes sense.

Reply:
I don't understand what you mean. 

If you mean that `tx_data` shouldn't be `const`, then the data from `st7789c_config` can't be transmitted directly. This could be solved by copying the data before calling transmit (as it did in the old code).

If you mean something else, could you explain how to solve your concerns?